google_compute_engine/instance_setup/instance_setup.py

```diff
@@ -48,22 +48,44 @@ def __init__(self, debug=False):
     self.metadata_dict = None
     self.instance_config = instance_config.InstanceConfig()
 
-    if self.instance_config.GetOptionBool(
-        'InstanceSetup', 'optimize_local_ssd'):
-      self._RunScript('optimize_local_ssd')
-    if self.instance_config.GetOptionBool('InstanceSetup', 'set_multiqueue'):
-      self._RunScript('set_multiqueue')
     if self.instance_config.GetOptionBool('InstanceSetup', 'network_enabled'):
       self.metadata_dict = self.watcher.GetMetadata()
+      self.instance_config = self._GetInstanceConfig()
       if self.instance_config.GetOptionBool('InstanceSetup', 'set_host_keys'):
         self._SetSshHostKeys()
       if self.instance_config.GetOptionBool('InstanceSetup', 'set_boto_config'):
         self._SetupBotoConfig()
+    if self.instance_config.GetOptionBool(
+        'InstanceSetup', 'optimize_local_ssd'):
+      self._RunScript('optimize_local_ssd')
+    if self.instance_config.GetOptionBool('InstanceSetup', 'set_multiqueue'):
+      self._RunScript('set_multiqueue')
     try:
       self.instance_config.WriteConfig()
     except (IOError, OSError) as e:
       self.logger.warning(str(e))
 
+  def _GetInstanceConfig(self):
+    """Get the instance configuration specified in metadata.
+
+    Returns:
+      string, the instance configuration data.
+    """
+    try:
+      instance_data = self.metadata_dict['instance']['attributes']
+    except KeyError:
+      instance_data = {}
+      self.logger.warning('Instance attributes were not found.')
+
+    try:
+      project_data = self.metadata_dict['project']['attributes']
+    except KeyError:
+      project_data = {}
+      self.logger.warning('Project attributes were not found.')
+
+    return (instance_data.get('instance-configs') or
```

Comment:
Great feedback. I've made a new pull request:
https://github.com/GoogleCloudPlatform/compute-image-packages/pull/411

I've changed the name to `google-instance-configs` and I've added error handling for both the metadata value, and for the config files. If the file cannot be parsed, it won't be used.